src/models/linear_network.py

```diff
@@ -0,0 +1,196 @@
+
+                        val_rmse.append(val_loss.item())
+
+            print(f'Epoch {epoch + 1}, train RMSE: {np.mean(train_rmse)}')
+
+            if early_stopping is not None:
+                epoch_val_rmse = np.mean(val_rmse)
+                print(f'Val RMSE: {epoch_val_rmse}')
+                if epoch_val_rmse < best_val_score:
+                    batches_without_improvement = 0
+                    best_val_score = epoch_val_rmse
+                else:
+                    batches_without_improvement += 1
+                    if batches_without_improvement == early_stopping:
+                        print('Early stopping!')
+                        return None
+
+    def predict(self) -> Tuple[Dict[str, Dict[str, np.ndarray]], Dict[str, np.ndarray]]:
+        test_arrays_loader = DataLoader(data_path=self.data_path, batch_file_size=self.batch_size,
+                                        shuffle_data=False, mode='test', to_tensor=True)
+
+        preds_dict: Dict[str, np.ndarray] = {}
+        test_arrays_dict: Dict[str, Dict[str, np.ndarray]] = {}
+
+        if self.model is None:
+            self.train()
+            self.model: LinearModel
+
+        self.model.eval()
+        with torch.no_grad():
+            for dict in test_arrays_loader:
+                for key, val in dict.items():
+                    preds = self.model(val.x)
+                    preds_dict[key] = preds.numpy()
+                    test_arrays_dict[key] = {'y': val.y.numpy(), 'latlons': val.latlons}
+
+        return test_arrays_dict, preds_dict
+
+
+class LinearModel(nn.Module):
+
+    def __init__(self, input_size, layer_sizes, dropout):
+        super().__init__()
+        layer_sizes.insert(0, input_size)
+
+        self.dense_layers = nn.ModuleList([
+            LinearBlock(in_features=layer_sizes[i - 1],
+                        out_features=layer_sizes[i], dropout=dropout) for
+            i in range(1, len(layer_sizes))
+        ])
+
+        self.final_dense = nn.Linear(in_features=layer_sizes[-1], out_features=1)
+
+        self.init_weights()
+
+    def init_weights(self):
+        for dense_layer in self.dense_layers:
+            nn.init.kaiming_uniform_(dense_layer.linear.weight.data)
+
+        nn.init.kaiming_uniform_(self.final_dense.weight.data)
+        # http://cs231n.github.io/neural-networks-2/#init
+        # see: Initializing the biases
+        nn.init.constant_(self.final_dense.bias.data, 0)
+
+    def forward(self, x):
+        # flatten
+        x = x.contiguous().view(x.shape[0], -1)
+        for layer in self.dense_layers:
+            x = layer(x)
+
+        return self.final_dense(x)
+
+
+class LinearBlock(nn.Module):
+    """
+    A linear layer followed by batchnorm, a ReLU activation, and dropout
+    """
+
+    def __init__(self, in_features, out_features, dropout=0.25):
+        super().__init__()
+        self.linear = nn.Linear(in_features=in_features, out_features=out_features, bias=False)
```

Comment:
The arguments to the linear model are just floats, not `torch.size`:

`input_size` is the size of the input data (which would be `num_timestamps * dimensions`, since everything is flattened - see the `LinearNetwork` class.

`layer_sizes` is just a list of floats of the sizes you want the interim linear layers to be (e.g. `[100, 50]`, where the input size will be the layer before's output size.

Reply:
So the right way to achieve the above is:

```python
In [59]: input_size = 55
    ...: layer_sizes = [55, 55, 55, 55, 55, 100]
    ...: droput = 0.25
    ...: LinearModel(input_size=input_size,
    ...:             layer_sizes=layer_sizes,
    ...:             dropout=dropout)
```
This would create a linear network with 6 linear blocks; the first 5 would have weight matrices of shape `[55, 55]`, and the last one would have shape `[55, 100]`.

In addition, a final linear layer with a weight matrix of shape `[100, 1]` will take the last vector and turn it into a prediction for vegetation health.